pkg/virt-api/rest/subresource.go

```diff
@@ -677,6 +677,32 @@ func (app *SubresourceAPIApp) UnfreezeVMIRequestHandler(request *restful.Request
 
 }
 
+func (app *SubresourceAPIApp) SoftRebootVMIRequestHandler(request *restful.Request, response *restful.Response) {
+
+	log.Log.Info("SoftRebootVMIRequestHandler")
+	validate := func(vmi *v1.VirtualMachineInstance) *errors.StatusError {
+		if vmi.Status.Phase != v1.Running {
+			return errors.NewConflict(v1.Resource("virtualmachineinstance"), vmi.Name, fmt.Errorf("VM is not running"))
+		}
+		condManager := controller.NewVirtualMachineInstanceConditionManager()
+		if condManager.HasConditionWithStatus(vmi, v1.VirtualMachineInstancePaused, v12.ConditionTrue) {
+			return errors.NewConflict(v1.Resource("virtualmachineinstance"), vmi.Name, fmt.Errorf("VMI is paused"))
+		}
+		if !condManager.HasCondition(vmi, v1.VirtualMachineInstanceAgentConnected) {
+			if !(*vmi.Spec.Domain.Features.ACPI.Enabled) {
```

Comment:
The `Features` field of the `DomainSpec` struct is optional. Probably it should be null-checked prior to dereferencing it. Same for the `Enabled` of the `FeatureState` (`ACPI` field).